pyo3-build-config/src/lib.rs

```diff
@@ -231,6 +224,22 @@ pub mod pyo3_build_script_impl {
     }
 }
 
+fn rustc_minor_version() -> Option<u32> {
+    static RUSTC_MINOR_VERSION: OnceCell<Option<u32>> = OnceCell::new();
+    RUSTC_MINOR_VERSION
+        .get_or_init(|| {
+            let rustc = env::var_os("RUSTC")?;
+            let output = Command::new(rustc).arg("--version").output().ok()?;
+            let version = core::str::from_utf8(&output.stdout).ok()?;
+            let mut pieces = version.split('.');
+            if pieces.next() != Some("rustc 1") {
+                return None;
+            }
+            pieces.next()?.parse().ok()
+        })
+        .clone()
```

Comment:
Clippy also appears to be unhappy about this `clone` call:

![grafik](https://github.com/PyO3/pyo3/assets/12997846/de02df16-1823-43f2-b978-463e7abda191)